src/main/java/pl/kzcwat/localincidentserver/region/RegionController.java

```diff
@@ -0,0 +1,106 @@
+package pl.kzcwat.localincidentserver.region;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.data.web.SortDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import pl.kzcwat.localincidentserver.region.exception.RegionNotFoundException;
+import pl.kzcwat.localincidentserver.region.request.RegionRequest;
+
+import java.net.URI;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/v1/region")
+public class RegionController {
+
+    private final RegionService regionService;
+
+    @GetMapping
+    public Page<RegionEntity> getRegionsPage(
+            @PageableDefault(page = 0, size = 5)
+            @SortDefault.SortDefaults({
+                    @SortDefault(sort = "regionId", direction = Sort.Direction.ASC)
+            })
+                    Pageable pageable) {
+        return regionService.getAllRegions(pageable);
+    }
+
+    @GetMapping("{regionId}")
+    public ResponseEntity<RegionEntity> getRegion(@PathVariable Long regionId) {
+        try {
+            Optional<RegionEntity> regionEntity = regionService.getRegion(regionId);
+            return regionEntity.map(value -> new ResponseEntity<>(value, HttpStatus.OK)).orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+        } catch (IllegalArgumentException e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        } catch (RegionNotFoundException e) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @PostMapping("addRegion")
```

Comment:
```suggestion
    @PostMapping
```

W restowym API nie powinniśmy nazywać endpointów z wykorzystaniem nazw z czasownikami, wystarczający jest czasownik HTTP POST i nazwa endpointa `/api/v1/region`.

Ponadto dodawany w nagłówku HTTP `Location` adres nowego zasobu jest tworzony z kontekstu requestu (`.fromCurrentRequest()`). Ponieważ ten endpoint ma dodane owe `addRegion`, to Location pojawia się niewłaściwie, np. `/api/v1/region/addRegion/1` zamiast `/api/v1/region/1`.